rs/rosetta-api/icp/tests/system_tests/test_cases/neuron_management.rs

```diff
@@ -288,3 +289,103 @@ fn test_start_and_stop_neuron_dissolve() {
         ));
     });
 }
+
+#[test]
+fn test_change_auto_stake_maturity() {
+    let rt = Runtime::new().unwrap();
+    rt.block_on(async {
+        let env = RosettaTestingEnvironment::builder()
+            .with_initial_balances(
+                vec![(
+                    AccountIdentifier::from(TEST_IDENTITY.sender().unwrap()),
+                    // A hundred million ICP should be enough
+                    icp_ledger::Tokens::from_tokens(100_000_000).unwrap(),
+                )]
+                .into_iter()
+                .collect(),
+            )
+            .with_governance_canister()
+            .build()
+            .await;
+
+        // Stake the minimum amount 100 million e8s
+        let staked_amount = 100_000_000u64;
+        let neuron_index = 0;
+        let from_subaccount = [0; 32];
+
+        env.rosetta_client
+            .create_neuron(
+                env.network_identifier.clone(),
+                &(*TEST_IDENTITY).clone(),
+                RosettaCreateNeuronArgs::builder(staked_amount.into())
+                    .with_from_subaccount(from_subaccount)
+                    .with_neuron_index(neuron_index)
+                    .build(),
+            )
+            .await
+            .unwrap();
+        env.pocket_ic.tick().await;
+
+        // See if the neuron was created successfully
+        let agent = get_test_agent(env.pocket_ic.url().unwrap().port().unwrap()).await;
+        let neuron = list_neurons(&agent).await.full_neurons[0].to_owned();
```

Comment:
I am not sure whether a refactoring makes a lot of sense here. I am first calling for the test agent which I am using in other parts of the code too, so I can't really refactor that. The list neurons call I could refactor but it is more coincidence that I am always interested in the first neuron an not any other consecutive neurons. Then again I would only extract a single line fo code into a function which would not make a lot of sense. The check for the state is always dependent on the part of the test so that can't really be refactored without making the function very specific (possibly only being called once per test scenario), which would defeat the purpose of refactoring code parts into a function.